app/services/record_declarations/base.rb

```diff
@@ -54,8 +54,10 @@ def call
         raise ActionController::ParameterMissing, errors.map(&:message)
       end
 
-      declaration = create_record!
+      create_declaration_attempt!
       validate_provider!
+      declaration = create_record!
+
```

Comment:
```suggestion
      declaration_attempt = create_declaration_attempt!
      validate_provider!
      declaration = create_record!
      declaration_attempt.update!(declaration: declaration)
```

Have you thought about whether we'd like to reference the declaration attempt back to the declaration itself? Would that be useful?